src/main/kotlin/com/exactpro/th2/http/server/HttpServer.kt

```diff
@@ -0,0 +1,230 @@
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.exactpro.th2.http.server
+
+import com.exactpro.th2.http.server.options.ServerOptions
+import com.exactpro.th2.http.server.util.LinkedData
+import mu.KotlinLogging
+import rawhttp.core.HttpVersion
+import rawhttp.core.RawHttp
+import rawhttp.core.RawHttpRequest
+import rawhttp.core.RawHttpResponse
+import rawhttp.core.body.BodyReader
+import java.io.IOException
+import java.lang.IllegalStateException
+import java.net.InetSocketAddress
+import java.net.ServerSocket
+import java.net.Socket
+import java.net.SocketException
+import java.util.UUID
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicBoolean
+
+class HttpServer(
+    private val options: ServerOptions,
+    private val terminationTime: Long,
+    socketDelayCheck: Long
+) : RawHttpServer {
+
+    @Volatile
+    private var active: AtomicBoolean = AtomicBoolean(false)
+    private val dialogManager: DialogueManager = DialogueManager(socketDelayCheck)
+
+    private var socket: ServerSocket = options.createSocket()
+    private val executorService: ExecutorService = options.createExecutorService()
+    private val additionalExecutors: ExecutorService = options.createExecutorService()
+    private val http: RawHttp = options.getRawHttp()
+
+    override fun start() {
+        if (active.get()) {
+            LOGGER.warn { "Server already started" }
+            return
+        }
+        active.set(true)
+
+        Thread({
+            LOGGER.info { "Server started" }
+            listen()
+        }, "th2-conn-http-server").start()
+
+        dialogManager.startCleaner()
+    }
+
+    private fun listen() {
+        while (active.get()) {
+            socket.runCatching(ServerSocket::accept)
+                .onFailure {
+                    if (active.get()) {
+                        when (it) {
+                            is SocketException -> {
+                                options.onError("Broken or closed server socket!", exception = it)
+                                recreateSocket()
+                            }
+                            else -> {
+                                options.onError("Failed to accept client socket", exception = it)
+                                if (socket.isClosed) recreateSocket()
+                            }
+                        }
+                    } else {
+                        LOGGER.warn(it) { "Server was stopped that leaded to errors: " }
+                    }
+                }.onSuccess { client ->
+                    executorService.submit {
```

Comment:
```suggestion
                    executorService.execute {
```
`execute` is prefferd over `submit` if waiting for result is not required because `execute` is usually a bit more lightweight